src/main/java/no/nav/arxaas/hierarchy/interval/IntervalBasedHierarchyBuilder.java

```diff
@@ -54,27 +55,40 @@ public IntervalBasedHierarchyBuilder(
     @Override
     public Hierarchy build(String[] column) {
         if(dataType == BuilderDataType.LONG){
-            HierarchyBuilderIntervalBased<Long> builder = arxHierarchyBuilderIntervalBased();
-            applyIntervals(builder);
-            applyLevels(builder);
-            builder.prepare(column);
-            return new Hierarchy(builder.build().getHierarchy());
+            return hierarchyFromlongTypeBuilder(column);
         }
         else if (dataType == BuilderDataType.DOUBLE){
-            HierarchyBuilderIntervalBased<Double> builder = arxHierarchyBuilderIntervalBasedDouble();
-            applyIntervalsDouble(builder);
-            applyLevels(builder);
-            builder.prepare(column);
-            return new Hierarchy(builder.build().getHierarchy());
+            return hierarchyFromdoubleTypeBuilder(column);
         }
         throw new IllegalStateException("Datatype=" + dataType.toString() + " is not supported");
     }
 
     /**
-     * Create HierarchyBuilderIntervalBased with right create method
+     * Create a HierarchyBuilderIntervalBased of type Double and add the intervals and levels to it
```

Comment:
can we pass the datatype as a parameter and make 2 methods into 1 that can handle either a Double or long??